src/isomorphic/React.js

```diff
@@ -86,4 +86,31 @@ var React = {
 
 };
 
+if (__DEV__) {
+  // React.DOM factories are deprecated. Set up a Proxy to listen for
+  // accesses to the React.DOM namespace and alert users to switch to
+  // the `react-addons-dom-factories` package.
+  if (typeof Proxy === 'function') {
+    var hasWarnedOfFactories = false;
```

Comment:
I really like this use of a `Proxy`. I'm wondering though about the fact that, [according to MDN, Proxy is not yet supported in IE](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Proxy#Browser_compatibility). To also show this warning in IE, we may want to switch and add the warning in the copy of `ReactDOMFactories` that remains in React. We could do that in a follow-up PR though.